docs/Supplementary Views.md

```diff
@@ -0,0 +1,76 @@
+
+Then, when you register a class or Nib for a supplementary view, you must specify the element kind that you are creating the registration for (whether you are using `UICollectionView.SupplementaryRegistration` or the manual `register` family of methods). You *must* have at least one registration for each element kind that your layout uses.
+
+When the collection view gets to the point where it needs to request a supplementary view for display, it will ask your data source to provide one for the given element kind it needs. For a collection view containing multiple types of supplementary views, each element kind will be requested separately. It is then your responsibility to look at the requested element kind and return a supplementary view for that element kind. It is entirely possible to have multiple `SupplementaryRegistration` or multiple re-use identifiers registered for a single element kind, but you must always ensure to match the requested element kind with the one that is returned.
+
+
+### How UtiliKit Helps
+
+UtiliKit aims to make this process as foolproof as possible. As always, there are 3 steps to correctly implementing the supplementary dequeue process- definition, registration, and dequeue. As an example, we'll walk through a case using a diffable data source and a compositional layout.
+
+1. Definition
+
+This occurs in the layout. In our case, when we are defining our `NSCollectionLayoutSection`. For example, a badge supplementary item on each of our cells may be defined something like this, where `UtiliKit` provides a small convenience here by providing some type-safety over the `String` implementation of `UIKit`:
+
+```swift
+extension UICollectionView.ElementKind {
+    static let badge = Self(rawValue: "badge-view")
+}
+
+func makeLayout() {
+  let badgeItem = NSCollectionLayoutSupplementaryItem(layoutSize: .init(widthDimension: .absolute(20), heightDimension: .absolute(20)),
+                                                      elementKind: .badge, containerAnchor: .init(edges: [.top, .leading]))
+
+  let itemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1),
+                                        heightDimension: .fractionalHeight(1))
+  let item = NSCollectionLayoutItem(layoutSize: itemSize,
+                                    supplementaryItems: [badgeItem])
+
+  // rest of layout definition
+}
+```
+
+2. Registration
+
+The next step is then registering the class(es) or nib(s) that will fulfill this element kind in the collection view. In this example, there is only one, that we register in `viewDidLoad`, ensuring we pass the same `UICollectionView.ElementKind` in to the registration.
+
+`collectionView.register(BadgeSupplementaryView.self, forSupplementaryViewOfKind: .badge)`
```

Comment:
Should we go ahead and wrap this in a "swift" code block to help with syntax highlighting when the markdown is rendered?
````suggestion
```swift
collectionView.register(BadgeSupplementaryView.self, forSupplementaryViewOfKind: .badge)
```
````

Reply:
Agreed, will fix.